### ISSUE TITLE
Static lighting on Corsair Strafe RGB

### ISSUE THREAD

**Issue by lenpup0** [roxaskeyheart/Chromatics#33]
Howdy folks, thanks for this very cool piece of software.

Chromatics is working and attaching to FFXIV correctly as far as I can tell. It's taking over CUE and I'm getting the base-state FFXIV lighting on my keyboard (blue with pink wasd, shift, crtl, space, and HP/MP/TP colors on F keys). 

Unfortunately that's all I'm getting. No change to red in combat/enmity gain, no hp/mp/tp changes reflected, no.. anything, but the base state.

EXCEPT! Once, and only once, the number keys changed to white from left to right in time with my castbar as I cast a teleport. I could not replicate this!

**Comment by lenpup0** [comment]
![image](https://user-images.githubusercontent.com/29714656/27558593-803a9af4-5a72-11e7-866b-ea3c83a49ad9.png)

**Comment by lenpup0** [comment]
![image](https://user-images.githubusercontent.com/29714656/27558624-a9234556-5a72-11e7-860e-8094d2eb9462.png)

**Comment by logicallysynced** [comment]
Will investigate this

**Comment by lenpup0** [comment]
After a few days of testing it seems clear that the macro errors are occurring at the end of my first cast of anything with a cast bar after logging in (or starting Chromatics). If I mount up, it 'crashes' in this way and the lighting freezes in its current state. If I was already mounted upon starting Chromatics, it crashes when I cast a spell after dismounting.
If I exit and restart Chromatics, I have normal function again until I cast a spell (anything with a castbar). I played around with it and found that the spell does not have to go off... if i move at the very end of the cast and the spell fizzles, the macro errors still occur and the lighting freezes. Possibly suggests it's related to the keyboard cast bar reaching the end, and not to do with whats happening in game.

I will test and see if, so long as I don't complete a castbar, the lighting keeps working or if anything else generates the macro errors.

Thanks again for your work on this awesome project. I hope it pays off for you above and beyond the satisfaction of creating something cool!

**Comment by logicallysynced** [comment]
If you open up the CUE2 software and open the Keyboard RGB emulator, and then unplug your keyboard, does the same thing happen?

Currently I don't have a physical Corsair device to test Chromatics with so we use the emulator to develop for Corsair devices. I haven't been able to replicate this issue currently so I was curious if it was directly related to the type of keyboard you use.

On that note what device model do you use?

**Comment by lenpup0** [comment]
When I opened CUE it was showing this, so I think I found the problem I need to correct before I can come back to you:
![image](https://user-images.githubusercontent.com/29714656/27742433-794ffb3c-5d6d-11e7-89e8-aa28f1f99ec6.png)

It's a Corsair STRAFE RGB

**Comment by lenpup0** [comment]
I have solved the "macro execution unavailable" issue within CUE by upgrading to Windows 10.

However Chromatics is still producing the same result after I cast anything (the keyboard castbar working correctly for this first cast only):
![image](https://user-images.githubusercontent.com/29714656/27754017-016552e0-5d9d-11e7-9318-a366d4367b0c.png)

Unplugging and replugging the keyboard after Chromatics throws the macro error causes the keyboard to go black, then Corsair default red, then black again.

Restarting Chromatics resumes FFXIV lighting. Then unplugging and replugging the keyboard causes it to go black, then to the Corsair default, then black again.

**Comment by logicallysynced** [comment]
Does the emulator run Chromatics when your keyboard is not plugged in?

Something along the lines of:

* Run FFXIV
* Open CUE2 Keyboard emulator
* Unplug physical keyboard
* Open Chromatics using mouse

**Comment by lenpup0** [comment]
Perplexingly, this time opening CUE2 shows "no device detected" despite my corsair being the device I'm typing with right now (and it being lit normally with default Corsair red & white lighting). 

No device errors in System -> Device Manager.
It was working prior to the last reboot!

I will troubleshoot and get back to you once I can test your requested steps. Sure hope I don't have a bad keyboard, its a week old.

**Comment by lenpup0** [comment]
~~Following the steps above, CUE2 just throws "no device detected!" error. Plugging in the keyboard after following those steps causes the STRAFE RGB to appear in CUE2, but Chromatics never kicks in, even tho the debug log shows all normal startup.~~
~~Is there a separate Emulator function? I cannot find one and the keyboard display in CUE disappears when I unplug the STRAFE.~~
~~Side note, what would be the lowest-priced Corsair RGB keyboard you could use for hands on testing? I'd kinda like to get one in your hands, but I'm not sure how to contact you directly.~~

I'm getting some oddly unpredictable results. I repeated the steps you mentioned and the emulated K95 RGB in CUE2 _did_ display the Chromatics lighting. This is after returning to the working 2.2.1 version.

**Comment by logicallysynced** [comment]
All the demo devices show up the top for me, it's possible you might need to uncheck "Show only connected devices" in the Advance Settings tab on CUE2?

And I certainly appreciate the thought and offer but you don't have to do that, the emulators are suppose to operate identical to the real thing according to Corsair themselves and we will be looking at purchasing some physical test devices in the future as well. :)

**Comment by logicallysynced** [comment]
I have been looking at this issue and have tried to apply some potential fixes. I believe the problem to be related to the fact that I have been testing on a K95 emulator (which has 18 macro keys) compared to yourself who has a Strafe RGB (which has 0 macro keys), so the null errors may be coming from Chromatics trying to set keys which don't exist.

The problem with this becomes that I have no way of testing since the emulator is a K95. Therefore I'm going to upload a pre-release of Version 2.2.3 here for you to test.

[Chromatics2.2.3aBETA-Installer.zip](https://github.com/roxaskeyheart/Chromatics/files/1117041/Chromatics2.2.3aBETA-Installer.zip)

**Comment by lenpup0** [comment]
~~Got it. Using K95 RGB Demo keyboard and following the recommended steps, the K95 RGB does not light up when I start Chromatics, with normal debug log and FFXIV attachment.~~ As before, when I plug the STRAGE RGB back in, I do not get Chromatics lighting, only my base default.
After testing again with 2.2.1 the Emulator DID work. 

~~Downloading new build now will get back to you!~~
No devices show in the new build.

**Comment by lenpup0** [comment]
New build does not show any devices:
![image](https://user-images.githubusercontent.com/29714656/27765524-6d92be3c-5e68-11e7-9d76-9401003ea9af.png)
 It was blank but contained a checkbox, so I clicked it. Clicking a check box creates a new blank row, I repeated 4x in the screenshot for the heck of it.

Blacked-out keyboard, normal debug log.

**Comment by logicallysynced** [comment]
Please try this

[Chromatics2.2.3bBETA-Installer.zip](https://github.com/roxaskeyheart/Chromatics/files/1117588/Chromatics2.2.3bBETA-Installer.zip)

**Comment by lenpup0** [comment]
![image](https://user-images.githubusercontent.com/29714656/27767603-7037861a-5eb1-11e7-9da3-67efb8325d09.png)

 This began before the load screen even ended, before the game world was visible or any keys were pressed.

I did note that lighting came on immediately in blue when I opened Chromatics, prior to launching FFXIV. That had not previously happened until I was almost at the end of the load screen to enter the game world.

**Comment by logicallysynced** [comment]
What happens if you start FFXIV first and then open Chromatics?

**Comment by lenpup0** [comment]
Launched Chromatics while @ Main Menu: keyboard went dark, no errors in debug log upon game world load, but keyboard remained dark. 
![image](https://user-images.githubusercontent.com/29714656/27767679-d32b275c-5eb3-11e7-9b92-d9cbc767510a.png)


Exited Chromatics and re-launched while already in game world: Behaved the same as in previous test when I ran it before FFXIV. Blue lighting, object reference error spamming debug log.
![image](https://user-images.githubusercontent.com/29714656/27767699-3c21d116-5eb4-11e7-884d-12dabf1014d1.png)

**Comment by logicallysynced** [comment]
And just to confirm, the devices were all enabled under the Devices tab yes?

**Comment by lenpup0** [comment]
All 4 are enabled although only a keyboard is plugged in.

**Comment by logicallysynced** [comment]
This version has a high level debug/logging level, I'm hoping it provides some more insight for me of where this issue is occurring as i cannot replicate it myself. Please copy paste the entire log window here.

[Chromatics2.2.3cBETA-Installer.zip](https://github.com/roxaskeyheart/Chromatics/files/1117665/Chromatics2.2.3cBETA-Installer.zip)

**Comment by lenpup0** [comment]
```
Starting Chromatics Version 2.2.2 (Beta)
Searching for devices.chromatics..
Attempting to load devices.chromatics..
devices.chromatics loaded.
Searching for settings.chromatics..
Attempting to load settings.chromatics..
settings.chromatics loaded.
Searching for mappings.chromatics..
Attempting to load mappings.chromatics..
mappings.chromatics loaded.
Attempting to load Razer SDK..
Architecture: x64
The Razer SDK (RzChromaSDK64.dll) Could not be found on this computer. Please install Synapse and restart ACT if you wish to use Chromatics with Razer devices.
Razer SDK failed to load.
Attempting to load Logitech SDK..
Logitech SDK failed to load.
Attempting to load CUE SDK..
CUE SDK Loaded (2.10.91/2.14.67)
Attempting to load LIFX SDK..
LIFX SDK Loaded
Waiting for Game Launch..
Waiting for Game Launch..
Waiting for Game Launch..
Waiting for Game Launch..
DX11 Initiated
Attached to FFXIV
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Main Menu is still active.
Game Running (Rhea Love)
Watchdog Started
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Watchdog Stopped
Game stopped
Init Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Init Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Init Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Init Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Init Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Init Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.

```

**Comment by lenpup0** [comment]
Also, the character name listed there is not mine?

**Comment by logicallysynced** [comment]
Yeah I've pinpointed a bug in the program now that I'm looking into. Hopefully this is the source of the issues you're having too.

**Comment by logicallysynced** [comment]
Please try the latest release (2.2.3)

https://github.com/roxaskeyheart/Chromatics/releases

**Comment by lenpup0** [comment]
`Internal` Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
``

**Comment by lenpup0** [comment]
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
Internal Error: Object reference not set to an instance of an object.

**Comment by logicallysynced** [comment]
Can you post a screenshot of the Chromatics install directory please?

**Comment by lenpup0** [comment]
![image](https://user-images.githubusercontent.com/29714656/27854342-51c9d9a0-611b-11e7-887b-f72a56d42f80.png)
![image](https://user-images.githubusercontent.com/29714656/27854355-6057e340-611b-11e7-8af3-ef86c97b9afc.png)

**Comment by logicallysynced** [comment]
Can you delete the following files and try re-running Chromatics?

- signatures-x64.json
- statuses.json
- structures-x64.json
- structures-x86.json

**Comment by lenpup0** [comment]
Same result on game world load:
Internal Error: Object reference not set to an instance of an object.
Parse Error: Object reference not set to an instance of an object.
repeating

**Comment by logicallysynced** [comment]
I feel this problem is going to forever elude me TT.TT haha

Here is the current development build I'm working on (2.2.4 pre-release), just want to make sure the same thing is happening on the same version I have.

[Chromatics2.2.4bBETA-Installer.zip](https://github.com/roxaskeyheart/Chromatics/files/1125301/Chromatics2.2.4bBETA-Installer.zip)

**Comment by lenpup0** [comment]
At game world load:

Internal Error:    at Chromatics.DeviceInterfaces.CorsairLib.ApplyMapKeyLighting(String key, Color col, Boolean clear)
   at Chromatics.Chromatics.GlobalApplyMapKeyLighting(String key, Color col, Boolean clear)
   at Chromatics.Chromatics.ProcessFFXIVData()
Parse Error: Object reference not set to an instance of an object.

Errors stop when I uncheck the box (un-enable keyboard) in Devices, lighting remains on (static).

I'm curious, did you add something that triggers on game world load in 2.2.2? Is that why I used to get my initial cast bar and then the errors, but now I get errors immediately on entering the world?

**Comment by logicallysynced** [comment]
Can you add me as a friend on Discord so we can look into this together? I know where the error is originating from but without a physical corsair device I have to keep sending you my incremental builds to test it. Communicating on Discord could make this process quicker and easier.

Roxas Keyheart#4950

**Comment by logicallysynced** [comment]
Resolved in 2.2.4
